motr/client_init.c

```diff
@@ -1678,16 +1678,36 @@ int m0_client_init(struct m0_client **m0c_p,
 	}
 
 	if (conf->mc_is_addb_init) {
-		char buf[64];
+		char        buf[256];
+		/* uint64 max character size */
+		enum { MAX_PID_IN_CHAR_SIZE = 20 };
+		const char *addb_stob_location = NULL;
 		/* Default client addb record file size set to 128M */
 		m0_bcount_t size = DEFAULT_CLIENT_ADDB2_RECORD_SIZE;
+
 		if (conf->mc_addb_size != 0) {
 			if (conf->mc_addb_size > MAX_ADDB2_RECORD_SIZE)
 				M0_LOG(M0_WARN, "ADDB size is more than recommended");
 			size = conf->mc_addb_size;
 			M0_LOG(M0_DEBUG, "ADDB size = %" PRIu64 "", size);
 		}
-		sprintf(buf, "linuxstob:./addb_%d", (int)m0_pid());
+#ifndef __KERNEL__
+		addb_stob_location = getenv("M0_CLIENT_ADDB_DIR");
+#endif
+		/* checking for buf size overflow */
+		if (addb_stob_location != NULL && sizeof(&addb_stob_location) >=
+		   sizeof(buf) - (sizeof("linuxstob:/addb_") +
+		   MAX_PID_IN_CHAR_SIZE))
+		{
+			M0_LOG(M0_WARN, "ADDB location is more than defined "
+			                "size .. ignoring it.");
+			addb_stob_location = NULL;
+		}
+		if (addb_stob_location == NULL)
+			addb_stob_location = ".";
+
+		snprintf(buf, 255, "linuxstob:%s/addb_%d",
+		         addb_stob_location, (int)m0_pid());
```

Comment:
Minor comment. It would be preferable to print the addb location under M0_DEBUG here.